simexpal/queuesock.py

```diff
@@ -63,28 +63,28 @@ def run(self):
 			if cur_subprocess is not None:
 				subprocess_terminated = cur_subprocess.poll() is not None
 
-			if subprocess_terminated and not len(requests) == 0:
-				request = requests.pop(0)
+			if subprocess_terminated:
+				if not len(requests) == 0:
+					request = requests.pop(0)
 
-				if self._should_stop:
+					if request['action'] == 'launch':
```

Comment:
I'd suggest to only push `launch` requests to the queue. Other requests can be handled immediately after they are received (not really related to the change here but we should probably refactor this as part of the PR).

Reply:
I removed the check ``if request['action'] == 'launch':``. Currently only ``launch`` requests are pushed to the queue anyway.